projects/fal/src/fal/toolkit/file/providers/fal.py

```diff
@@ -64,6 +85,23 @@ def _upload_file(self, upload_url: str, file: FileData):
         with urlopen(req):
             return
 
+    def gc_futures(self):
+        import traceback
+
+        for future in self.uploads.copy():
+            if not future.done():
+                continue
+
+            if future in self.uploads:
+                self.uploads.remove(future)
```

Comment:
ATL: why the if? if it's for race conditions, shouldn't we do a better locking mechanism or handle KeyError exception?